lalrpop/src/lexer/dfa/overlap.rs

```diff
@@ -69,10 +75,18 @@ fn add_range(range: Test, start_index: usize, disjoint_ranges: &mut Vec<Test>) {
             let max_max = cmp::max(range.end(), overlapping_range.end());
             // When working with inclusive ranges, we need to be sure to not double count
             // the meeting points of low-mid_range and mid-max_range.
-            // So we subtract 1 from the end of the low_range and mid_range as these elements are already included in the start of their corresponding next ranges.
-            let low_range = Test::new(min_min..=mid_min - 1);
-            let mid_range = Test::new(mid_min..=mid_max - 1);
-            let max_range = Test::new(mid_max..=max_max);
+            // So we adjust the end of the low_range and start of max_range as these elements are already included in the start of their corresponding next ranges.
+
+            let low_range = if mid_min == 0 {
+                // This is an edgecase where both ranges start at the null character
+                // In this case we don't want to create a range from 0 to -1
+                // Thus we create an empty range
+                Test::new(RangeInclusive::new(1, 0))
+            } else {
+                Test::new(min_min..=mid_min - 1)
+            };
```

Comment:
```suggestion
            // This is an edgecase where both ranges start at the null character
            // In this case we don't want to create a range from 0 to -1
            // Thus we create an empty range
            Test::new(min_min..=mid_min.checked_sub(1))
```
Is this possible? I don't know well about this code. So please check which version express the design intent better.

Reply:
`checked_sub` returns an option right? So you need to handle the `None` case. I guess one could write something like 

```rust
Test::new(mid_min.checked_sub(1).map_or_else(||RangeInclusive::new(1, 0), |v| min_min..=v))
```

I'm not sure if that's cleaner?

Reply:
oh, i am sorry. I was thinking of `saturated_sub` 😂

Reply:
https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_sub

Reply:
Currently, if mid_min would underflow, I am returning an empty range which will get ignored later. With saturating_sub, we would end up with a valid range of `\0`..=`\0`. I think this would mean having a duplicate value, probably `\0`..=`\0` twice.